### ISSUE TITLE
should ui.freeze/thaw do anything when called with no arguments

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1426]
This was found when working in #1425 

The existing documentation for the `sherpa.ui.freeze` and `sherpa.ui.thaw` routines (which are used by the `sherpa.astro.ui` layer) suggest that when called with no arguments they do something (ie change all the models).

In fact, they don't. They do nothing. So in #1425 I updated the documentation (after adding tests of the current behavior). I believe this is the sensible thing to do. However, I wanted to create this issue to ask if we actually wanted the code to behave as the documentation used to say - namely (for `freeze`)

> If called with no arguments, then all parameters of models in source expressions are frozen.

This would not be terribly hard to do, but I don't think it's worth doing as it is easier for a user to mess things up here, rather than  `freeze()` just doing nothing.

We **COULD** make `freeze` and `thaw` require at least one argument. This seems like the better approach.

If we did want to make `freeze()` work then it would be something like (after #1425 has landed):

```
for id in self.list_models_id():
    self.get_source(id).freeze()
```

**Comment by hamogu** [comment]
I agree that just fixing the docs to match that code as opposed to fixing the code to match the docs is the way to go, and @DougBurke suggestion of just requiring one argument is a great way that users don't have code that they expect to do something when in fact is does not.

Thawing everything is confusing once one has multiple datasets and multiple models. The extra convenience to save some typing is not worth that risk in my opinion.
